### DISCUSSION TITLE
CI/CD Using doks, Gitlab, and pandoc

### DISCUSSION THREAD

**Discussion by atwriter** [thuliteio/doks#232] [Q&A]
Howdy all. I have been using Hugo and the doks theme for a site (thanks again for the great theme and great support here!). There has been a recent need to start converting some of the md files published in hugo and doks to PDF. My tool of choice here is using pandoc w/ latex for the md to pdf conversion. 

My end goal is to have a fully automated CI/CD that includes building the doks theme site along with pdf conversions in one pipeline. The pdf will be hosted and downloadable right from the output.

I am just getting started on this but am wondering if anyone else has already accomplished this? I supposed I am in need of a decent yaml script for gitlab. 

Any tips or help is appreciated!

**Comment by epogrebnyak** [comment]
What is the source of your pdfs? The markdown content files? Or the build html files?

**Reply by atwriter** [reply]
The markdown files, for starters. But, if it is easier, it can be the html files.

**Reply by epogrebnyak** [reply]
Sorry, not any special kind of magic that I personally have at disposal. There are some CI workflows that seem to do conversion like https://github.com/marketplace/actions/create-pdf have you checked? Or you need Gitlab in specific?

**Reply by epogrebnyak** [reply]
We have a Russian-speaking documentation authoring group - I think I'd accept a well-formulated question (something you have tried and why it does not work) in English too: https://t.me/docsascode
